http-api/src/patches.rs

```diff
@@ -52,9 +100,145 @@ async fn patches_handler(ctx: Context, urn: Urn) -> Result<impl Reply, Rejection
                 tracing::warn!("Failed to resolve identities in patch {}: {}", id, e);
             }
 
+            let revisions = resolve_revisions(&patch, &mut browser, &meta, storage.as_ref());
+
+            let patch = patch::Patch {
+                author: patch.author,
+                title: patch.title,
+                state: patch.state,
+                target: patch.target,
+                labels: patch.labels,
+                timestamp: patch.timestamp,
+                revisions,
+            };
+
             Cob::new(id, patch)
         })
         .collect();
 
     Ok(warp::reply::json(&all))
 }
+
+async fn patch_handler(
+    ctx: Context,
+    urn: Urn,
+    patch_id: ObjectId,
+) -> Result<impl Reply, Rejection> {
+    let repo = git::Repository::new(ctx.paths.git_dir()).map_err(Error::from)?;
+    let storage = ctx.storage().await?;
+    let project = identities::project::get(storage.as_ref(), &urn)
+        .map_err(Error::Identities)?
+        .ok_or(Error::NotFound)?;
+    let meta: project::Metadata = project.try_into().map_err(Error::Project)?;
+
+    let whoami = person::local(&*storage).map_err(Error::LocalIdentity)?;
+    let store = Store::new(whoami, &ctx.paths, &storage).map_err(Error::from)?;
+    let patches = patch::PatchStore::new(&store);
+    let mut patch = patches
+        .get(&urn, &patch_id)
+        .map_err(Error::from)?
+        .ok_or(Error::NotFound)?;
+    if let Err(e) = patch
+        .resolve(storage.as_ref())
+        .map_err(Error::IdentityResolve)
+    {
+        tracing::warn!("Failed to resolve identities in patch {}: {}", patch_id, e);
+    }
+
+    let mut browser = git::Browser::new_with_namespace(
+        &repo,
+        &git::Namespace::try_from(urn.encode_id().as_str()).map_err(|_| Error::MissingNamespace)?,
+        git::Branch::remote(
+            meta.default_branch.as_str(),
+            &patch.author.peer.default_encoding(),
+        ),
+    )
+    .map_err(Error::from)?;
+
+    let revisions = resolve_revisions(&patch, &mut browser, &meta, storage.as_ref());
+
+    let patch = patch::Patch {
+        author: patch.author,
+        title: patch.title,
+        state: patch.state,
+        target: patch.target,
+        labels: patch.labels,
+        timestamp: patch.timestamp,
+        revisions,
+    };
```

Comment:
Since you're moving out of `Patch`, I would pass it into `resolve_revisions` by value, and return it. That way you don't need to clone all the fields like you do bellow.

Reply:
I think you would have to use `into_iter` as well

Reply:
Yes, but you avoid cloning the entire discussion :)